pkg/boot/linux.go

```diff
@@ -0,0 +1,147 @@
+package boot
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/u-root/u-root/pkg/cpio"
+	"github.com/u-root/u-root/pkg/kexec"
+	"github.com/u-root/u-root/pkg/uio"
+)
+
+// LinuxImage implements OSImage for a Linux kernel + initramfs.
+type LinuxImage struct {
+	Kernel  io.ReaderAt
+	Initrd  io.ReaderAt
+	Cmdline string
+}
+
+func newLinuxImageFromArchive(a *archive) (*LinuxImage, error) {
+	kernel, ok := a.Files["modules/kernel/content"]
+	if !ok {
+		return nil, fmt.Errorf("kernel missing from archive")
+	}
+
+	li := &LinuxImage{}
+	li.Kernel = kernel
+
+	if params, ok := a.Files["modules/kernel/params"]; ok {
+		b, err := ioutil.ReadAll(uio.Reader(params))
+		if err != nil {
+			return nil, err
+		}
+		li.Cmdline = string(b)
+	}
+
+	if initrd, ok := a.Files["modules/initrd/content"]; ok {
+		li.Initrd = initrd
+	}
+	return li, nil
+}
+
+func NewLinuxImage(kernel, initrd io.ReaderAt, cmdline string) *LinuxImage {
```

Comment:
I'm not sure this function is really useful, since LinuxImage is exported and has all its fields exported

Reply:
Good point, dropped it.